#FleisKappa

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [36]:
import os

path = "/content/drive/MyDrive/Gemastik/mergeData"
os.listdir(path)

['labeled_data_yuma.csv', 'labaled_data_rehan.csv', 'labeled_data_ameng.csv']

In [37]:
import pandas as pd
import os
from statsmodels.stats.inter_rater import fleiss_kappa, aggregate_raters



df_yuma = pd.read_csv(os.path.join(path, 'labeled_data_yuma.csv')).iloc[0:100]
df_rehan = pd.read_csv(os.path.join(path, 'labaled_data_rehan.csv')).iloc[0:100]
df_ameng = pd.read_csv(os.path.join(path, 'labeled_data_ameng.csv')).iloc[0:100]


df_kappa = pd.DataFrame({
    'Yuma': df_yuma['Stance'],
    'Rehan': df_rehan['Stance'],
    'Ameng': df_ameng['Stance']
})

df_kappa = df_kappa.astype(str).apply(lambda x: x.str.strip().str.lower())


data_agregat,categories = aggregate_raters(df_kappa.values)
kappa = fleiss_kappa(data_agregat, method='fleiss')

print(f"Skor Fleiss' Kappa: {kappa:.4f}")

Skor Fleiss' Kappa: 0.9119


**CONCLUSION**

Kami melakukan pembahasan kesepakatan dengan cara 100 baris pertama di ketiga data kami mempunyai value yang sama di kolom "Teks Komentar" dan sisa nya mempunyai data yang berbeda dari ketiga masing data. Hal ini dilakukan untuk sampling perhitungan Fleiss Kappa, dan setelah perhitungan dilakukan Fleiss Kappa menunjukan skor 0.91 yang menunjukan kita sudah melakukan sampling yang cukup konsisten